src/ophyd_async/epics/areadetector/writers/nd_plugin.py

```diff
@@ -10,13 +10,43 @@ class Callback(str, Enum):
     Disable = "Disable"
 
 
+class ADBaseDataType(int, Enum):
+    Int8 = 0
+    UInt8 = 1
+    Int16 = 2
+    UInt16 = 3
+    Int32 = 4
+    UInt32 = 5
+    Int64 = 6
+    UInt64 = 7
+    Float32 = 8
+    Float64 = 9
```

Comment:
I'd prefer this to be `str, Enum` and go on the strings given by EPICS rather than the indexes